Tasks/AzureResourceManagerTemplateDeploymentV3/operations/DeploymentScopeBase.ts

```diff
@@ -18,7 +19,27 @@ export class DeploymentScopeBase {
     }
 
     public async deploy(): Promise<void> {
-        await this.createTemplateDeployment();
+        try {
+            await this.createTemplateDeployment();
+        } catch (error) {
+            if((error as string).toLowerCase().indexOf("serviceprincipal") != -1) {
```

Comment:
is it not possible to have a better string to detect - 'serviceprincipal' is too generic. Will there not be an error code or status or something which indicates 403 or access denied or forbidden etc.
Can you please point me to a sample failure.

Reply:
I've updated to check for status code. 

Sample Failure:
2019-12-04T10:00:50.9956997Z ##[error]The client '94405baa-4126-47a1-baeb-e5ae9c4151e1' with object id '94405baa-4126-47a1-baeb-e5ae9c4151e1' does not have authorization to perform action 'Microsoft.Resources/deployments/write' over scope '/subscriptions/c00d16c7-6c1f-4c03-9be1-6934a4c49682/resourcegroups/issacnitinrg/providers/Microsoft.Resources/deployments/azuredeploy-20191204-153050-957f' or the scope is invalid. If access was recently granted, please refresh your credentials.